projects/orquestra-workflow-shared/Makefile

```diff
@@ -0,0 +1,129 @@
+################################################################################
+# © Copyright 2021-2024 Zapata Computing Inc.
+################################################################################
+
+# Use just "python" as the interpreter for all make tasks. It will use your
+# virtual environment if you activate it before running make. You can override
+# the interpreter path like:
+# make test PYTHON=/tmp/other/python/version
+PYTHON="python"
+
+clean:
+	find . -regex '^.*\(__pycache__\|\.py[co]\)$$' -delete;
+	find . -type d -name __pycache__ -exec rm -r {} \+
+	find . -type d -name '*.egg-info' -exec rm -rf {} +
+	find . -type d -name .mypy_cache -exec rm -r {} \+
+	rm -rf .pytest_cache;
+	rm -rf tests/.pytest_cache;
+	rm -rf dist build
+	rm -f .coverage*
+
+
+test:
+	$(PYTHON) -m pytest \
+		--durations=10 \
+		tests
+
+
+# Min code-test coverage measured for the whole project required for CI checks to pass.
+MIN_COVERAGE=0
+
+
+# Option explanation:
+# - '--cov=src' - turn on measuring code coverage. It outputs the results in a
+#    '.coverage' binary file. We're not using it, but it can be input to other
+#    tools like 'python -m coverage report'
+# - '--cov-report xml' - in addition, generate an XML report and store it in
+#    coverage.xml file. It's required to upload stats to codecov.io.
+coverage:
+	$(PYTHON) -m pytest \
+		--cov=src \
+		--cov-fail-under=$(MIN_COVERAGE) \
+		--cov-report xml \
+		--no-cov-on-fail \
+		--durations=10 \
+		tests \
+		&& echo Code coverage Passed the $(MIN_COVERAGE)% mark!
+
+# Reads the code coverage stats from '.coverage' file and prints a textual,
+# human-readable report to stdout.
+show-coverage-text-report:
+	$(PYTHON) -m coverage report --show-missing
+
+BASE_BRANCH := $(if $(GITHUB_BASE_REF),$(GITHUB_BASE_REF),main)
+github-actions-coverage-report:
+	@$(PYTHON) -m coverage report --show-missing
+	@$(PYTHON) -m diff_cover.diff_cover_tool coverage.xml --compare-branch=origin/$(BASE_BRANCH)
+
+github_actions:
```

Comment:
```suggestion
github-actions:
```

Reply:
Same as in previous comment, this has to be consistent across all 3 packages and Ill rather leave it as it is for now